### ISSUE TITLE
Formatting not working in Neovim

### ISSUE THREAD

**Issue by GaetanLepage** [Myriad-Dreamin/tinymist#290]
**Describe the bug**

Editor: neovim
Formatting does not seem to work although I have set `formatterMode` to `"typstyle"`.
By _"formatting"_, I mean the capability of the language server to respond to `textDocument/formatting` requests as sent by `vim.lsp.buf.format()`.

I get the following message in the `lsp.log` file:
```
[WARN][2024-05-14 00:25:18] ...-format/init.lua:173     '"textDocument/formatting" is not supported for tinymist, not attaching lsp-format'
```

**Package/Software version:**
Tynimist v0.11.8 (install with nix)

**Comment by Myriad-Dreamin** [comment]
is there any extra thing to do like tinymist have to be setup in nvim lsp, which is probably the cause? Tinymist was not configured anything specific to lsp formatting for neovim.

**Comment by GaetanLepage** [comment]
> is there any extra thing to do like tinymist have to be setup in nvim lsp, which is probably the cause? Tinymist was not configured anything specific to lsp formatting for neovim.

I don't think that it is a configuration issue.
Are you sure that the language server itself implements the [`textDocument/formatting`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_formatting) API ?

**Comment by Myriad-Dreamin** [comment]
Definitely, it's been working in vscode for months. https://docs.rs/lsp-types/0.95.1/src/lsp_types/request.rs.html

**Comment by GaetanLepage** [comment]
Ok I see. My guess is that the dynamic registration of `textDocument/formatting` is not occuring for me.
Is having the `typstyle` binary necessary for `tinymist` to format a document or does it use a typstyle "library" that is shipped with tinymist ?

**Comment by GaetanLepage** [comment]
Well, I guess that I got my answer https://github.com/neovim/nvim-lspconfig/issues/2807.
Are you OK to leave this issue opened ? Neovim 0.10 should be released in a couple of days. I will then try and close the issue if it works.
Of course, you may also want to close it now as this is definitely not a bug in tinymist.

**Comment by Myriad-Dreamin** [comment]
I can improve the documentation for it, that it requires neovim 0.10.

**Comment by GaetanLepage** [comment]
Well, apprently you can even make it work with neovim 0.9: https://www.reddit.com/r/neovim/comments/1ao6c5a/comment/kpx2s8v/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by GaetanLepage** [comment]
> Is having the `typstyle` binary necessary for `tinymist` to format a document or does it use a typstyle "library" that is shipped with tinymist ?

So, no need for additional dependencies ? Tinymist is basically _ready to format_ out of the box ?

**Comment by Myriad-Dreamin** [comment]
> > Is having the `typstyle` binary necessary for `tinymist` to format a document or does it use a typstyle "library" that is shipped with tinymist ?
> 
> So, no need for additional dependencies ? Tinymist is basically _ready to format_ out of the box ?

You don't have to install formatters. tinymist integrates both typstyle and typstfmt, as they have quite small size of binary (in 1~10 KBs). If we are lucky, we just need to configure `formatterMode` to `"typstyle"` to make it work.

**Comment by GaetanLepage** [comment]
> You don't have to install formatters. tinymist integrates both typstyle and typstfmt, as they have quite small size of binary (in 1~10 KBs). If we are lucky, we just need to configure `formatterMode` to `"typstyle"` to make it work.

Great !
Thank you for the help and clear explanation !

**Comment by Andrew15-5** [comment]
> Neovim 0.10 should be released in a couple of days.

Wait, really? That's huge!

**Comment by GaetanLepage** [comment]
It works fine with nvim 0.10 !

**Comment by GaetanLepage** [comment]
For the record, even though `vim.lsp.buf.format()` now works fine, the [lsp-format.nvim](https://github.com/lukas-reineke/lsp-format.nvim/) plugin is still refusing to attach because `client.supports_method('textDocument/formatting')` returns `false`. Later on, it does return `true`...
Anyway, I made a [PR](https://github.com/lukas-reineke/lsp-format.nvim/pull/87) to add a `force_attach` option that skips this check.

**Comment by liangz678** [comment]
> It works fine with nvim 0.10 !

My tinymist format still doesn't work on Nvim10.0. Do I need any additional settings?

**Comment by mawkler** [comment]
I'm having the same issue. `=vim.lsp.buf.format()` yields the error `[LSP] Format request failed, no matching language servers`.

I'm using [the configuration suggested in the docs](https://myriad-dreamin.github.io/tinymist//frontend/neovim.html#loc-1x20.00x716.93):

```lua
require("lspconfig")["tinymist"].setup {
    settings = {
        formatterMode = "typstyle",
        exportPdf = "onType",
        semanticTokens = "disable"
    }
}
```
